### DISCUSSION TITLE
Istio Ingress Gateway Not Receiving the original Source IP of the request

### DISCUSSION THREAD

**Discussion by gabbar023** [istio/istio#48756] [Q&A]
We have created k8s clusters using 5 vms where we have used keepalive to create a floating VIP.We are using this VIP as the istio LB IP.
My issue is All the requests coming to this Istio LB, we are not receiving the original source ip of the request. The source ip is getting overriden by the Tunnel Interface IP of the server where this VIP is attached.
How can i get the original source ip and avoid this Tunnel interface IP overriden for istio gateway?
Can someone pls help?

**Comment by abasitt** [comment]
If you are sending traffic directly to the VIP without any other LB fronting it, it's most likely modified by the kube-proxy. You can try few options e.g.

1. CNI specific setting like DSR if you are using Cilium OR
2. externalTrafficPolicy local. https://kubernetes.io/docs/tutorials/services/source-ip/#source-ip-for-services-with-type-loadbalancer

**Comment by rootsongjc** [comment]
Your question is similar to [this](https://github.com/istio/istio/discussions/47640), you can check out the [Forwarding external client attributes (IP address, certificate info) to destination workloads](https://istio.io/latest/docs/ops/configuration/traffic-management/network-topologies/#forwarding-external-client-attributes-ip-address-certificate-info-to-destination-workloads).

**Comment by gabbar023** [comment]
i added this annotation in the istio-ingress template, 
proxy.istio.io/config: '{"gatewayTopology" : { "numTrustedProxies": 3,"proxyProtocol":
          {}  } }'
 I'm receving the actual source ip as well as the K8s Tunnel Interface ip also, 
 So i'm still not able to block the actual source ip as istio is considering the tunnel ip as source ip,
 How do i ignore the tunnel created for the LB  IP being captured by istio ingress gateway

**Comment by gabbar023** [comment]
[2024-01-11T04:14:32.978Z] "GET /favicon.ico HTTP/1.1" 200 - via_upstream - "-" 0 34494 2 2 "133.237.92.81,240.0.151.192" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36"

So for my all requests, the tunnel IP  240.0.151.192 is getting attached, how do i ignore this?

**Reply by abasitt** [reply]
AFAIK the first IP (the left most) should take preference. how are you blocking the source ip? can you share the auth policy you are trying ?

**Comment by rootsongjc** [comment]
I recently wrote a blog post titled [Maintaining Traffic Transparency: Preserving Client Source IP in Istio](https://tetrate.io/blog/istio-source-ip-transparency/) that addresses this topic in detail. I hope you find it helpful in solving your source IP preservation challenges. Feel free to check it out and let me know if you have any questions or feedback.

**Comment by share-me** [comment]
If it helps, see this related thread: https://github.com/istio/istio/issues/58295#issuecomment-3772179778
